### PR TITLE
HAAR-2668: add deployment to cloud platform prod env

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: helm_deploy/values-prod-cp.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+# Per environment values which override defaults in nomis-user-roles-api/values.yaml
+
+generic-service:
+  replicaCount: 4
+
+
+  ingress:
+    host: nomis-user-roles-api-preprod.prison.service.justice.gov.uk
```
Comment (ravmoj) [inline helm_deploy/values-prod-cp.yaml:9]: reference of preprod

File: helm_deploy/values-prod-cp.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    dso_enabled: false
+    tlsSecretName: hmpps-nomis-user-roles-api-cert
+
+    allowlist:
+      groups:
+        - internal
+
+  env:
+    APPLICATIONINSIGHTS_CONFIGURATION_FILE: applicationinsights.dev.json
```
Comment (petergphillips) [inline helm_deploy/values-prod-cp.yaml:18]: wrong app insights file?

File: .circleci/config.yml [hunk 3]
```diff
@@ -111,6 +111,20 @@ workflows:
             - nomis-user-roles-api-prod
           requires:
             - request-prod-approval
+      - hmpps/deploy_env:
+          name: deploy_prod-cp
+          env: "prod-cp"
+          jira_update: false
```
Comment (ravmoj) [inline .circleci/config.yml:117]: jira_update ,slack_notification are false here

Comment (petergphillips) [reply .circleci/config.yml:117]: not sure need it enabling for both prod environments though

Comment (SimonMitchellMOJ) [reply .circleci/config.yml:117]: that is already updated with the deployment to azure - when the full swap is done will set to true

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**